### PR TITLE
electron 31.7.6

### PR DESCRIPTION
**Pull request by Dexus**
- updated to electron v31.7.6

### CODE REVIEW

File: patches/electron-settings.patch
Comment (ShaMan123) [inline patches/electron-settings.patch:1]: I wouldn't go with patching. I managed to dump the dep while keeping the functionality.

File: main.js [hunk 1]
```diff
@@ -72,43 +73,44 @@ if (!gotTheLock) {
...
   mainWindow = new BrowserWindow({
-    width: Math.ceil(width*0.9), 
-    height: Math.ceil(height*0.9), 
-    frame: !frameless, 
+    width: Math.ceil(width * 0.9),
+    height: Math.ceil(height * 0.9),
+    frame: !frameless,
     show: false,
     webPreferences: {
       contextIsolation: false,
```
Comment (ShaMan123) [inline main.js:93]: I thought this had to be true after electron@19...

**Comment by ShaMan123** [comment]
@Dexus I see all your PRs do not conflict with my fork... I can rebase on top of it and then if you wish you/we can PR it into upstream (as long as @abebeos is up for the work)

**Comment by abebeos** [comment]
finally seeing electron 33, thanks a lot!

> * little bit styling (auto format)

As a general MUST rule, you need to keep whitespace changes in separate PRs, or (at minimum) within a separate commit (exclusively for whitespace). This is to simplify review.

This PR would become like this:

* whitespace
* init electron 33.2.1
* replace deprecated request with axio + node detection
* (alternatively the whitespace changes/commit at the end)

@ShaMan123 , once Dexus stuff is merged, I'll give you both more access-rights, so you can review/merge by yourselves.

(In things go nirvana, I can still revert stuff back to last functionality)

**Comment by abebeos** [comment]
@Dexus , my prefered way would be that you provide

* PR electron 33
* PR some more fixes
* PR whitespace

@Dexus & @ShaMan123 , I assigned you admin rights to the repo. You should be able now to review/merge by yourselves. Though you should have gotten direct-write access, too, please use the usual workflow (PR). If you prefer, you can work on branches on this repo, too (then the usual PR).

Just please stick to the basics:

* don't overdo if with adding too much stuff in one PR/commit
* whitespace & formatting: exclusively, in of PR (or at least in one commit)
* think of exhausted reviewers

**Comment by Dexus** [comment]
Do we have any more C++ living developers on hand? 

I'm surprised that it always falls flat on its face here... Only on Windows, but since I don't know C++ I'm stuck here. Theoretically, it should be enough if we don't build the plugin directly here in the project, but provide it as a real plugin. Which we can then import into Electron and also at NodeJ level.

If I'm not completely wrong now, and I'm running 33.2.1 on Windows at least... until the day before yesterday and I restarted the PC... Otherwise we would have to go back to electron v3x. Why is electron doing their own stupid thing again anyway?

**Comment by Dexus** [comment]
Okay, since electon use a never v8 engine with v32 + v33 then nodejs use, it will not compile. So i moved it back to the v31. Which is working. And a note for myself: `don't build electron yourself again O_o this makes things working for you but not in the wild`.

**Comment by Dexus** [comment]
@ShaMan123 @abebeos if you like to review, I will create the other PRs based on this one.

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+    "editor.formatOnSave": false,
+    "editor.formatOnPaste": false
+}
```
Comment (abebeos) [inline .vscode/settings.json:4]: editor settings are kept usualy out of the repo.

**Review by abebeos — APPROVED** [review]
approving visually, not tested locally with the packaged app.

File: main.js (hunk 1 above)
Comment (abebeos) [reply main.js]: @Dexus , don't leave review-comments unresolved.

**Comment by abebeos** [comment]
> Do we have any more C++ living developers on hand?

yes. Just file a new issue.

**Comment by abebeos** [comment]
@Dexus , delete the vscode settings file, and add it to gitignore, then

* 'Rebase and merge' this PR yourself from the github UI (curious to see if the access-rights work)

**Comment by Dexus** [comment]
Will first on Monday again ready have from tomorrow a short vacation

**Comment by abebeos** [comment]
> Will first on Monday again ready have from tomorrow a short vacation

Have fun!!!

**Review by ShaMan123 — APPROVED** [review]
We can introduce the test file I wrote that executes the app or introduce playwright if it works with v31 on linux (v19 didn't) so we have a baseline...

**Pull request merged**